src/core/agent/agent.ts

```diff
@@ -166,6 +166,15 @@ class Agent {
     return this.instance;
   }
 
+  set signifyClientInstance(signifyClient: SignifyClient) {
+    this.connectionService.signifyClientInstance = signifyClient;
+    this.credentialService.signifyClientInstance = signifyClient;
+    this.identifierService.signifyClientInstance = signifyClient;
+    this.ipexCommunicationService.signifyClientInstance = signifyClient;
+    this.multiSigService.signifyClientInstance = signifyClient;
+    this.signifyNotificationService.signifyClientInstance = signifyClient;
+  }
```

Comment:
I'm not sure why this is necessary - if we are using dependency injection by passing the signifyclient via agentServiceProps - shouldn't it be the same class instance that is already passed through? So auto update when we reconnect.

Reply:
@iFergal When initDatabase, we need init instance service, but signifyclient = undefined. We must set signifyclient to the service after start or bootAndConnect